opm/grid/GraphOfGridWrappers.cpp

```diff
@@ -594,6 +597,152 @@ zoltanPartitioningWithGraphOfGrid(const Dune::CpGrid& grid,
+            GraphOfGrid gog(grid, transmissibilities, edgeWeightMethod);
+            addWellConnections(gog, wellConnections);
+            gog.addNeighboringCellsToWells(layers);
+
+            // call partitioner
+            setGraphOfGridZoltanGraphFunctions(zz, gog, partitionIsEmpty);
+            rc = Zoltan_LB_Partition(zz, /* input (all remaining fields are output) */
+                                     &changes, /* 1 if partitioning was changed, 0 otherwise */
+                                     &numGidEntries, /* Number of integers used for a global ID */
+                                     &numLidEntries, /* Number of integers used for a local ID */
+                                     &numImport, /* Number of vertices to be sent to me */
+                                     &importGlobalGids, /* Global IDs of vertices to be sent to me */
+                                     &importLocalGids, /* Local IDs of vertices to be sent to me */
+                                     &importProcs, /* Process rank for source of each incoming vertex */
+                                     &importToPart, /* New partition for each incoming vertex */
+                                     &numExport, /* Number of vertices I must send to other processes*/
+                                     &exportGlobalGids, /* Global IDs of the vertices I must send */
+                                     &exportLocalGids, /* Local IDs of the vertices I must send */
+                                     &exportProcs, /* Process to which I send each of the vertices */
+                                     &exportToPart); /* Partition to which each vertex will belong */
+            numImport = 0;
+            if (rc == ZOLTAN_OK) {
+                gIDtoRank.resize(grid.numCells(), root);
+                for (int i = 0; i < numExport; ++i) {
+                    gIDtoRank[exportGlobalGids[i]] = exportToPart[i];
+                }
+                // partitioner sees only one cell per well, modify remaining
+                extendGIDtoRank(gog, gIDtoRank, root);
+
+                // prepare export lists for communication, exclude root's own cells
+                exportedCells = makeExportListsFromGIDtoRank(gIDtoRank, cc.size());
+                std::vector<int> rootCells;
+                exportedCells[root].swap(rootCells);
+                myImportList.reserve(rootCells.size());
+                for (const auto& cell : rootCells) {
+                    myImportList.emplace_back(cell, root, static_cast<char>(AttributeSet::owner), -1);
+                }
+            }
+        }
+        Zoltan_LB_Free_Part(&exportGlobalGids, &exportLocalGids, &exportProcs, &exportToPart);
+        Zoltan_LB_Free_Part(&importGlobalGids, &importLocalGids, &importProcs, &importToPart);
+        Zoltan_Destroy(&zz);
+    }
+
+    cc.broadcast(&rc, 1, root);
+    if (rc != ZOLTAN_OK)
+        OPM_THROW(std::runtime_error, "Could not initialize Zoltan!");
```

Comment:
Depending on what fails the error message is wrong. Isit  possible to have a different message for each failure?